ion/src/device/boot/rt0.cpp

```diff
@@ -75,3 +75,9 @@ void start() {
 
   abort();
 }
+
+volatile long millis_elapsed = 0;
+
+void __attribute__((interrupt)) sysTick() {
```

Comment:
Are you sure this is required? GCC's doc says "On ARMv7-M the interrupt type is ignored, and the attribute means the function may be called with a word aligned stack pointer."

Reply:
You're right, it is not required. There is a bit more information in the [ARM infocenter](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0774g/vvi1465810473505.html)

> In ARMv6-M, ARMv7-M, and ARMv8-M, the architectural exception handling mechanism preserves all processor registers, and a standard function return can cause an exception return. Therefore, specifying the interrupt attribute does not affect the behavior of the compiled output. However, ARM recommends using the interrupt attribute on exception handlers for clarity and easier software porting.